src/js/angular/clustermanagement/directives/cluster-legend.directive.js

```diff
@@ -0,0 +1,153 @@
+import {ClusterSvgService, HexagonConfig, LineConfig} from "../services/cluster-svg.service";
+
+const modules = [];
+angular.module('graphdb.framework.clustermanagement.directives.cluster-legend', modules)
+    .directive('clusterLegend', clusterLegend);
+
+clusterLegend.$inject = ['$rootScope', '$sce'];
+
+function clusterLegend($rootScope, $sce) {
+    return {
+        restrict: 'E',
+        templateUrl: 'js/angular/clustermanagement/templates/cluster-legend.html',
+        scope: {
+            clusterViewSelector: '='
+        },
+        link: ($scope, element, attrs) => {
+            // =========================
+            // Public variables
+            // =========================
+            $scope.nodeStates = [];
+            $scope.syncStatuses = [];
+            $scope.linkStates = [];
+
+            // =========================
+            // Private variables
+            // =========================
+            const legendElement = element.find('.cluster-legend-container');
+
+            // =========================
+            // Public functions
+            // =========================
+            $scope.toggleLegend = () => {
+                legendElement.toggleClass('hidden');
+            };
+
+            $scope.closeDialog = () => {
+                legendElement.addClass('hidden');
+            };
+
+            // =========================
+            // Private functions
+            // =========================
+            const getNodeStates = () => {
+                return [
+                    {
+                        labelKey: 'cluster_management.cluster_graphical_view.node_state_leader',
+                        svg: $sce.trustAsHtml(ClusterSvgService.getLeaderSvg(new HexagonConfig().setSize(30).setClasses('node leader')))
+                    },
+                    {
+                        labelKey: 'cluster_management.cluster_graphical_view.node_state_follower',
+                        svg: $sce.trustAsHtml(ClusterSvgService.getLeaderSvg(new HexagonConfig().setSize(30).setClasses('node follower')))
```

Comment:
Why is this called `getLeaderSvg`, are svgs differernt for leader, follower or out of sync nodes?

Reply:
I renamed it to getHexagonSvg, but it seems to have been lost. I'll rename it again.

Reply:
It might be `getNodeSvg` to be more readable. After all it is the Cluster SVG Service